### DISCUSSION TITLE
useIsFetching for multiple queryKeys

### DISCUSSION THREAD

**Discussion by nguyenyou** [TanStack/query#3339] [Q&A] [answered]
Say I have two useQuery hooks

```jsx
const {data: videos} = useQuery('videos', getVideos)
const {data: suggestions} = useQuery('suggestions', getSuggestions)
```

Instead of combining their `isLoading` manually, for example:

```jsx
const {data: videos, isLoading: isLoadingVideos} = useQuery('videos', getVideos)
const {data: suggestions, isLoading: isLoadingSuggestions} = useQuery('suggestions', getSuggestions)

const isLoading = isLoadingVideos || isLoadingSuggestions
```

Can we do this?

```jsx
const isFetching = useIsFetching(['videos', 'suggestions'])
```

**Comment by TkDodo** [comment, marked as answer]
you can use the `predicate` option:

```
useIsFetching({ predicate: query => ['videos', 'suggestions'].includes(query.key) })
```

but keep in mind that there is no fuzzy matching when doing it like that. Alternatively, you can call `useIsFetching` twice and sum up the results.

**Reply by nguyenyou** [reply]
Hmm..so if I have 5 queries, I have to call useIsFetching 5 times and sum up the results 🤔 probably what I'm trying to get rid of. Using predicate looks really close to what I'm trying to achieve. Thank you.
